daemon/main_test.go

```diff
@@ -49,18 +115,58 @@ var _ = g.BeforeSuite(func() {
 })
 
 var _ = g.Describe("Main", func() {
-	g.Context("Connection", func() {
-		dummyPluginDPU := NewDummyPlugin()
-		DpuDaemon := NewDpuDaemon(dummyPluginDPU)
+	var (
+		tmpDir           string
+		err              error
+		serverSocketPath string
+		dpuDaemon        *DpuDaemon
+		hostDaemon       *HostDaemon
+	)
+	g.BeforeEach(func() {
+		tmpDir, err = os.MkdirTemp("", "cniserver")
+		Expect(err).NotTo(HaveOccurred())
+		serverSocketPath = filepath.Join(tmpDir, "server.socket")
 
+		dummyPluginDPU := NewDummyPlugin()
+		dpuDaemon = NewDpuDaemon(dummyPluginDPU)
 		dummyPluginHost := NewDummyPlugin()
-		HostDaemon := NewHostDaemon(dummyPluginHost)
+		m := SriovManagerStub{}
+		hostDaemon = NewHostDaemon(dummyPluginHost).
+			WithCniServerPath(serverSocketPath).
+			WithSriovManager(m)
+	})
+
+	g.AfterEach(func() {
+		klog.Info("Cleaning up")
+		os.RemoveAll(tmpDir)
+		dpuDaemon.Stop()
+		hostDaemon.Stop()
+	})
 
-		g.It("Should connect succesfully if the server is up first", func() {
-			HostDaemon.Start()
-			DpuDaemon.Start()
-			_, err := HostDaemon.CreateBridgePort(1, 1, 1)
-			Expect(err).ShouldNot(HaveOccurred())
+	g.Context("Connection", func() {
+		g.It("Should connect successfully if the DPU Daemon is up first", func() {
+			dpuListen, err := dpuDaemon.Listen()
+			Expect(err).NotTo(HaveOccurred())
+			go func() {
+				err = dpuDaemon.Serve(dpuListen)
+				Expect(err).NotTo(HaveOccurred())
+			}()
+
+			hostListen, err := hostDaemon.Listen()
+			Expect(err).NotTo(HaveOccurred())
+			go func() {
+				_ = hostDaemon.Serve(hostListen)
```

Comment:
Same as above we should check the return `err`?

Reply:
We should not because serve can fail and that's expected.

Reply:
But serve is not allowed to fail for the DPU Daemon?